### PR TITLE
FDN-2397:Upgrade scala v2.13.13 and sbt version upgrade

### PR DESCRIPTION
**Pull request by abhidwivedi2701**
In this PR we are upgrading scala version from v2.13.10 to v2.13.13 and sbt version from v1.9.9 to v1.10.0 for which need to add type annotation in the below files

 api/app/actors/TaskActor.scala
 api/test/util/DependencySpec.scala
 www/app/lib/Permissions.scala

### CODE REVIEW

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
Click here for code coverage report: https://jenkins.flo.pub/job/flowcommerce/job/dependency/job/PR-758/6/scoverage-report/

File: api/app/actors/TaskActor.scala [hunk 1]
```diff
@@ -194,7 +194,7 @@ class TaskActorSyncLibrariesByPrefix @Inject() (
   ) {
   override def accepts(task: InternalTask): Boolean = {
     task.data match {
-      case _: TaskActorSyncLibrariesByPrefix => true
+      case _: TaskDataSyncLibrariesByPrefix => true
```
Comment (gheine) [inline api/app/actors/TaskActor.scala:197]: Why was this necessary?

**Review by gheine — CHANGES_REQUESTED** [review]
see above

File: api/app/actors/TaskActor.scala (hunk 1 above)
Comment (gheine) [reply api/app/actors/TaskActor.scala:197]: Oh that was actually wrong and the updated compiler helped to catch it! Amazing!

**Review by gheine — APPROVED** [review]

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Review by gheine — APPROVED** [review]

**Comment by flow-tech** [comment]
helm diff detected no changes for dependency-api

helm diff detected no changes for dependency-www

**Comment by flow-tech** [comment]
Click here for code coverage report: https://jenkins.flo.pub/job/flowcommerce/job/dependency/job/PR-758/20/scoverage-report/

**Review by gheine — APPROVED** [review]

**Review by arias-m — APPROVED** [review]

**Pull request merged**